Modified Introduction

dmwheel1 opened the pull request:
Made detailed changes to the intro to clearly define role of TEE and not introduce unnecessary topics too early. Added additional clarity to the problems the protocol must address

Commit:
Modified Introduction

Made detailed changes to the intro to clearly define role of TEE and not introduce unnecessary topics too early. Added additional clarity to the problems the protocol must address

hannestschofenig committed:
Minor editorial changes

Commit:
Merge pull request #1 from hannestschofenig/master

Minor editorial changes

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -83,38 +89,83 @@ https://github.com/teep/teep-architecture-spec. Instructions are on that
+TEE. Typically, applications are chosen to execute inside a TEE because
+those applications perform security sensitive operations or operate on
+sensitive data. An application running inside a TEE is referred to as a 
+Trusted Applications (TA), while a normal application running in the 
```
"Applications" should be "Application"

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -83,38 +89,83 @@ https://github.com/teep/teep-architecture-spec. Instructions are on that
+even greater concerns.
+
+The Trusted Execution Environment (TEE) concept is designed to execute
+applications in a protected environment that separates applications
```
For SGX, I find it odd to call the part in the enclave an "application", especially if it's a DLL which is a "library".  In my review I suggested a generic term like "application component" instead of "application" when defining a TA (I'm fine with the TA term)

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -83,38 +89,83 @@ https://github.com/teep/teep-architecture-spec. Instructions are on that
+But not all TEEs are the same, and different vendors may have different
+implementations of TEEs with different security properties, different
+features, and different control mechanisms to operate on TAs. Some
+vendors may themsleves market multiple different TEEs with different
```
typo "themselves"

mingpeiwk replied:
Noted.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -83,38 +89,83 @@ https://github.com/teep/teep-architecture-spec. Instructions are on that
+implementations of TEEs with different security properties, different
+features, and different control mechanisms to operate on TAs. Some
+vendors may themsleves market multiple different TEEs with different
+properties atuned to different markets. A device vendor may integrate
```
typo "atuned" -> "attuned"

mingpeiwk replied:
Will update.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -83,38 +89,83 @@ https://github.com/teep/teep-architecture-spec. Instructions are on that
+with TAs in a TEE, an interoperable protocol for managing TAs running in
+different TEEs of various devices is needed. In this TEE ecosystem,
+there often arises a need for an external trusted party to verify the
+identity, claims, and rights of SPs, devices, and their TEEs. This
```
"SPs" acronym is used before being defined

mingpeiwk replied:
To be updated.

dthaler commented on draft-ietf-teep-architecture.md:
```diff
@@ -83,38 +89,83 @@ https://github.com/teep/teep-architecture-spec. Instructions are on that
+identity, claims, and rights of SPs, devices, and their TEEs. This
+trusted third party is the Trusted Application Manager (TAM).   
+
+This protocol addresses the following problems:
```
"This protocol" would be referring to OTrP but this is the arch doc, so either "This architecture" (my preference), or at least "The protocol"

mingpeiwk replied:
Will update this to "This architecture"

dthaler reviewed (changes requested):
Overall, changes are good

Commit:
Modifications to RoT and Trust Anchor Defn

Updated definitions to align better with TCG understandings and NIST definitions

dmwheel1 commented:
Need to resolve how we merge this in, and who does it. I don't have permissions.

mingpeiwk reviewed (commented):
Will accept and merge these changes, and updates few changes noted.

mingpeiwk commented:
Merge the changes, and prepare to fix for the suggestion from the comments.
